ui/src/app/controllers/views/Controllers.tsx

```diff
@@ -28,12 +28,7 @@ const Controllers = (): JSX.Element => {
         controllersURLs.controller.usbDevices(null, true),
         controllersURLs.controller.vlans(null, true),
       ].map((path) => (
-        <Route
-          exact
-          key={path}
-          path={path}
-          render={() => <ControllerDetails />}
-        />
+        <Route exact path={path} render={() => <ControllerDetails />} />
```

Comment:
Removed the key prop as react-router handles this internally and we were overwriting the key it's using causing route to re-render.

Reply:
I can't remember for sure, but I wonder if the key was there so that if you change tabs it doesn't maintain the same scroll position. It might be worth checking.

Reply:
I can't imagine why this would break it, the order of the array never changes so even if a default array index was used as a key, the render still should work as expected. I checked it anyway and haven't noticed anything strange.

Reply:
but honestly why did RRv6 have to go and break everything :(

Reply:
Hmm... it does also appear to break the scrolling for me when it doesn't have the key:

![Screen Recording 2022-06-07 at 6 01 10 pm](https://user-images.githubusercontent.com/361637/172328461-93a61d75-14c1-4ce5-9053-93d0ec1d9a21.gif)

Reply:
ok thanks, at least I now know what you meant exactly with the scrolling. I'll double check.

Reply:
@huwshimi  It's not caused by this change, looks like we forgot to add the `useScrollToTop` hook to `Controllers`

Reply:
I think what's happening is when using dynamic `key`s we're overriding `Route`'s intended behaviour, causing the component to unmount and remount on every route change. This unmount/remount triggers `get` and `setActive` calls, but it also sends the scroll position back to the top. So I don't think the current implementation breaks scrolling so much as fixes an unintentional scroll to top. We should probably be more explicit about when we want route changes to automatically scroll to the top, i.e. using `useScrollOnRender()` or something like that

Reply:
> We should probably be more explicit about when we want route changes to automatically scroll to the top, i.e. using `useScrollOnRender()` or something like that

Can you think of a use case where the current implementation via `useScrollToTop` (scroll to top on pathname change) would not be sufficient?

If you think of each new pathname like a new page this pretty much replicates regular browser's behaviour (with static HTML pages).

Reply:
Nope! `useScrollToTop` works fine